### DISCUSSION TITLE
Pass updates

### DISCUSSION THREAD

**Discussion by ianbale** [alexandercerutti/passkit-generator#66] [Ideas]
Do you have any plans to add support for updating passes?

I have a new use case to dynamically update the barcode and other info, so just starting to look into this.

**Comment by alexandercerutti** [comment]
No plans actually. I had the idea to add an example of creating a pass using `node-apn` or such, but I don't have any plans, and actually the intention, of adding such support.

The reason is quickly explained: this library can be used to generate a pass, so a zip file with `.pkpass`.
From what I always understood, pass updates basically consists of just generating a new pass and sending it as a reply to Apple Wallet or who else for it.

Therefore this library can be used whenever you want.
Take the official resource as reference: https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/PassKit_PG/Updating.html

You (server) send the notification.
Wallet asks you for changes, and you reply to it with updated serial numbers and then with the updated pass.

Of course, there might be something that could be improved to make it easier to generate passes in such a situation, but until now, I never heard of anyone asking for changes.

Also, not everyone wants to generate a pass and update it. It depends.
So I decided to allow anyone to implement such a feature on its own if needed.
For example: if you run the server on a lambda, you don't need the code for the server.
If you would have it, it would have brought you to have longer startup times on cold starts due to dependencies probably.

I'm always open to discuss for changes and improvements. ✌

**Comment by ianbale** [comment]
Thanks for the info. I will take a look at this. It's not my highest priority work at the moment, writing Google Pay Passes has just been agreed with the train companies, so that is going to be pretty important for me to do soon.

**Comment by ianbale** [comment]
I've implemented this now and have updating passes ready to go live. As you correctly pointed out, there isn't much, if anything that you could roll into the library to help.

**Reply by alexandercerutti** [reply]
What you might do is to perform a sort of code-analysis and understand if some of the proposed changes in other Discussions' topics might improve your situation or see if anything might get improved and let me know. Not mandatory, of course 😄
